runtime/lua/vim/lsp/rpc.lua

```diff
@@ -407,7 +407,7 @@ function Client:handle_body(body)
   end
```

Comment:
Well, I think instead of such (repetitive) nil checks, we should handle the invalid response and throw early. LSP is not supposed to return a NIL as a RPC response, so if `decoded == nil` then we should raise an error as early as here.